src/libstd/ffi/c_str.rs

```diff
@@ -245,6 +245,9 @@ impl CString {
 
     /// Transfers ownership of the string to a C caller.
     ///
+    /// If the `CString` is empty, the returned pointer address will be
+    /// meaningless.
```

Comment:
For `CString` and `CStr` the pointer is definitely not meaningless as even if the string is empty it will still point to a single zero byte.